src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.EVP.Cipher.cs

```diff
@@ -13,12 +14,74 @@ internal static partial class Crypto
         [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_EvpCipherCreate2")]
         internal static extern SafeEvpCipherCtxHandle EvpCipherCreate(
             IntPtr cipher,
-            byte[] key,
+            ref byte key,
             int keyLength,
             int effectivekeyLength,
-            byte[] iv,
+            ref byte iv,
             int enc);
 
+        [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_EvpCipherCreatePartial")]
+        internal static extern SafeEvpCipherCtxHandle EvpCipherCreatePartial(
+            IntPtr cipher);
+
+        [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_EvpCipherSetKeyAndIV")]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        private static extern bool EvpCipherSetKeyAndIV(
+            SafeEvpCipherCtxHandle ctx,
+            ref byte key,
+            ref byte iv,
+            EvpCipherDirection direction);
+
+        internal static void EvpCipherSetKeyAndIV(
+            SafeEvpCipherCtxHandle ctx,
+            ReadOnlySpan<byte> key,
+            ReadOnlySpan<byte> iv,
+            EvpCipherDirection direction)
+        {
+            if (!EvpCipherSetKeyAndIV(
+                ctx,
+                ref MemoryMarshal.GetReference(key),
+                ref MemoryMarshal.GetReference(iv),
+                direction))
+            {
+                throw CreateOpenSslCryptographicException();
+            }
+        }
+
+        [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_EvpCipherGcmAvailable")]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        internal static extern bool EvpCipherGcmAvailable();
+
+        [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_EvpCipherCcmAvailable")]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        internal static extern bool EvpCipherCcmAvailable();
+
+        [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_EvpCipherSetGcmNonceLength")]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        private static extern bool EvpCipherSetGcmNonceLength2(
```

Comment:
What's the `2`?  A disambiguator?  Just name it the full name: `CryptoNative_EvpCipherSetGcmNonceLength`